### PR TITLE
Panel API update

### PR DESCRIPTION
**Pull request by deini**
`<Panel>` now accepts a `header` and `action` prop.

### CODE REVIEW

File: packages/docs/PropTables/PanelPropTable.tsx [hunk 1]
```diff
@@ -1,12 +1,14 @@
...
 
 export const PanelPropTable: React.FC = () => (
-  <Text>
-    A <Code>Panel</Code> can use all <Code>Box</Code> props, besides <Code primary>backgroundColor</Code>,{' '}
-    <Code primary>shadow</Code>, and any <Code primary>padding</Code> props. Otherwise, supports all native{' '}
-    <Code>&lt;div /&gt;</Code> element attributes.
-  </Text>
+  <PropTable>
+    <PropTable.Prop name="header" types="string">
+      Defines the panel header text
```
Comment (jorgemoya) [inline packages/docs/PropTables/PanelPropTable.tsx]: We've been using periods after the description.

File: packages/docs/PropTables/PanelPropTable.tsx [hunk 2]
```diff
@@ -1,12 +1,14 @@
...
+    </PropTable.Prop>
+    <PropTable.Prop name="action" types="ButtonProps &amp; { text: string }">
+      Defines the panel action button
```
Comment (jorgemoya) [inline packages/docs/PropTables/PanelPropTable.tsx]: Same as above.

**Comment by deini** [comment]
@jorgemoya addressed in [commit](https://github.com/bigcommerce/big-design/compare/b706000ee042d15d8786e3dcaf101738b216fc96..6d6d2115d0dff02c9b1c3833dc6ac16b597540d1)

File: packages/big-design/src/components/Panel/Panel.tsx [hunk 3]
```diff
@@ -1,19 +1,78 @@
...
+
+    if (typeof header !== 'string') {
+      return null;
+    }
+
+    return (
+      <Flex justifyContent="space-between" flexDirection="row">
+        {header && (
+          <Flex.Item>
+            <H2>{header}</H2>
```
[the remark this answers is not in the document]
Comment (deini) [reply packages/big-design/src/components/Panel/Panel.tsx]: Yep

File: packages/big-design/src/components/Panel/Panel.tsx [hunk 4]
```diff
@@ -1,19 +1,78 @@
...
+          </Flex.Item>
+        )}
+        {action && (
+          <Flex.Item>
+            <Button {...action}>{action.text}</Button>
```
[the remark this answers is not in the document]
Comment (deini) [reply packages/big-design/src/components/Panel/Panel.tsx]: Not at the moment according to design.

File: packages/big-design/src/components/Panel/Panel.tsx [hunk 5]
```diff
@@ -1,19 +1,78 @@
...
 import { StyledPanel } from './styled';
 
-export type PanelProps = React.HTMLAttributes<HTMLElement> & MarginProps;
-
-export const RawPanel: React.FC<PanelProps> = props => (
-  <StyledPanel
-    marginBottom="medium"
-    {...props}
-    backgroundColor="white"
-    shadow="raised"
```
[the remark this answers is not in the document]
Comment (deini) [reply packages/big-design/src/components/Panel/Panel.tsx]: We are still using it 🤔 am I missing something? 😱

[the remark this answers is not in the document]
Comment (chanceaclark) [reply packages/big-design/src/components/Panel/Panel.tsx]: Never mind, I confused myself.

File: packages/big-design/src/components/Panel/Panel.tsx (hunk 3 above)
[the remark this answers is not in the document]
Comment (chanceaclark) [reply packages/big-design/src/components/Panel/Panel.tsx]: This is something we need to bring up with design, but Form legends do not have consistency with stylings from any of our headers. If we are going to be using forms within panels, especially with `big-design-jsonschema-form`, we should make legends consistent with one of our headings.

**Pull request merged**